File: leetcode_C++/p1375.cpp
/*
@Author: mrxhar
@Date:   2023-09-11 11:11:00
@Email:  1337904145@qq.com
@Blog:   https://leetcode.cn/problems/number-of-times-binary-string-is-prefix-aligned/description/?envType=daily-question&envId=2023-09-11
*/
#include <vector>
using std::vector;

class Solution {
public:
    int numTimesAllBlue(vector<int>& flips) {
        int maxnum = 0, res = 0;
        for (int i = 0; i < flips.size(); i++) {
            maxnum = std::max(flips[i], maxnum);
            if (maxnum == i + 1) {
                res++;
            }
        }
        return res;
    }
};

File: leetcode_py/p1375.py
from typing import List


class Solution:
    def numTimesAllBlue(self, flips: List[int]) -> int:
        # 前缀一致， i 前面的数都为 1
        maxnum = 0
        res = 0
        for i, num in enumerate(flips):

            maxnum = max(maxnum, num)
            if maxnum <= i + 1:
                # 这时候 我们 ＋ 1
                res = max(res, res + 1)

        return res
